src/components/button/Etaps.tsx

```diff
@@ -1,84 +1,138 @@
 import { useState, useEffect } from "react";
-import { database } from "../../utils/firebase/firebase.config.jsx";
-import {
-  collection,
-  getDocs,
-  DocumentData,
-  query,
-  where,
-  onSnapshot,
-} from "firebase/firestore";
-import Activites from "../activities/Activities";
-
-interface IEtap {
+import { database } from "../../utils/firebase/firebase.config"; // import storage
+import { collection, getDocs } from "firebase/firestore";
+import { ref, getDownloadURL, getStorage } from "firebase/storage";
+import { getApp } from "firebase/app";
+import Activities from "../activities/Activities";
+
+interface Etap {
   id: string;
   name: string;
   sort: number;
+  icon: string;
 }
 
 function Etaps() {
-  const [etaps, setEtaps] = useState<IEtap[]>([]);
+  const [etaps, setEtaps] = useState<Etap[]>([]);
   const [etapId, setEtapId] = useState<string | null>(null);
-  const [isButtonDisabled, setIsButtonDisabled] = useState<boolean>(false);
+  const [activitiesByEtap, setActivitiesByEtap] = useState<
+    Record<string, { etap_id: string; id: string }[]>
```

Comment:
co to za element " Record"?

Reply:
Record jest TS, i jest to obiekt,na którym będę wykonywał reduce, i kluczami tego obiektu są stringi. Mam taki obiekt {etap_id: string, id: string}. Potrzenuję tego stanu do stworzenia tablicy etapid i aktywności, a potem porównania z wykonanymi przez usera aktywnościami